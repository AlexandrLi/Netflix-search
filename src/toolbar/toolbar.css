.container {
  background-color: #F5F5F5;
  display: flex;
  justify-content: space-between;
  padding: 20px 0;
}

.container div {
  margin-left: 100px;
}

.row {
  display: flex;
  justify-content: space-between;
  padding: 0 20px;
  margin-right: 100px;
}

.row div {
  margin: 0 10px;
}