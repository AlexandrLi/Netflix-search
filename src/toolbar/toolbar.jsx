import React from 'react';
import s from './toolbar.css';

export class Toolbar extends React.Component {

  handleClick(order) {
    this.props.onSortClick(order);
  }

  render() {
    return (
      <div className={s.container}>
        <div>
          {this.props.count} movies found
        </div>
        <div className={s.row}>
          Sort By
          <div className={this.props.sortBy === 'release' ? s.selected : ''} onClick={() => this.handleClick('release')}>
            release date
          </div>
          <div className={this.props.sortBy === 'rating' ? s.selected : ''} onClick={() => this.handleClick('rating')}>
            rating
          </div>
        </div>
      </div>
    )
  }
}