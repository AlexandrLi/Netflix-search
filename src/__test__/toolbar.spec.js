import React from "react"
import renderer from "react-test-renderer"
import { shallow } from "enzyme";
import { Toolbar } from "../toolbar/toolbar";

describe("Toolbar snapshot", () => {
  it("Take toolbar snapshot", () => {
    const renderedToolbar = renderer.create(<Toolbar count={10} sortBy={"release"} onSortClick={() => null} />).toJSON();
    expect(renderedToolbar).toMatchSnapshot();
  })
});

const mockFn = jest.fn();

describe("Shalow render of toolbar component", () => {
  let component;
  beforeEach(() => {
    component = shallow(<Toolbar count={10} sortBy={"release"} onSortClick={mockFn} />)
  });

  it("Component successfully rendered", () => {
    expect(component.length).toEqual(1);
  });

  it("contains div with movies count", () => {
    expect(component.find('div').get(1).props.children.join("")).toBe("10 movies found");
  });

  it("triggers handleClick function", () => {
    component.find('.selected').simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });
});