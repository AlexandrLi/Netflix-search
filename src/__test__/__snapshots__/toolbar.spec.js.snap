// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar snapshot Take toolbar snapshot 1`] = `
<div
  className="container"
>
  <div>
    10
     movies found
  </div>
  <div
    className="row"
  >
    Sort By
    <div
      className="selected"
      onClick={[Function]}
    >
      release date
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      rating
    </div>
  </div>
</div>
`;
